.github/workflows/rust.yml

```diff
@@ -0,0 +1,137 @@
+name: Rust CI
+
+on:
+  push:
+  pull_request:
+  schedule:
+    - cron: "0 0 * * *"
```

Comment:
do we need to run actions in this file on a schedule as well?

Reply:
The idea behind this workflow being run on a schedule is to continuously test the codebase against updates to Rust nightly and dependencies.